src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c

```diff
@@ -21,47 +47,574 @@ const ActorInit Bg_Po_Event_InitVars = {
+        blockPushDist = 0.0f;
+        this->dyna.actor.speedXZ = 0.0f;
+        this->direction = 5;
+        blockPosAtRest++;
+        this->actionFunc = BgPoEvent_BlockStop;
+        if (this->type == 1) {
+            return;
+        }
+        BgPoEvent_BlockCheck(this);
+        BgPoEvent_BlockCheck((BgPoEvent*)this->dyna.actor.parent);
+    }
+    func_8002F974(&this->dyna.actor, NA_SE_EV_ROCK_SLIDE - SFX_FLAG);
+}
+
+void BgPoEvent_BlockReset(BgPoEvent* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (this->dyna.unk_150 != 0.0f) {
+        player->stateFlags2 &= ~0x10;
+        this->dyna.unk_150 = 0.0f;
+    }
+    if (Math_ApproxF(&this->dyna.actor.posRot.pos.y, 493.0f, 1.0f) &&
+        Math_ApproxUpdateScaledS(&this->dyna.actor.shape.rot.z, this->dyna.actor.posRot.rot.z - 0x4000, 0x400)) {
+
+        this->index = (this->index + 1) % 4;
+        this->actionFunc = BgPoEvent_BlockFall;
+        sPuzzleState = 0;
+        if (this->type == 1) {
+            this->timer += 10;
+            this->timer = CLAMP_MAX(this->timer, 120);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Event/BgPoEvent_Init.s")
+void BgPoEvent_BlockSolved(BgPoEvent* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Event/BgPoEvent_Destroy.s")
+    if (this->dyna.unk_150 != 0.0f) {
+        player->stateFlags2 &= ~0x10;
+    }
+    if (Math_ApproxF(&this->dyna.actor.posRot.pos.y, 369.0f, 2.0f)) {
+        sPuzzleState = 0x20;
+        Actor_Kill(&this->dyna.actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Event/func_808A68D0.s")
+void BgPoEvent_AmyWait(BgPoEvent* this, GlobalContext* globalCtx) {
+    if (this->collider.base.acFlags & 2) {
+        sPuzzleState |= 0x20;
+        this->timer = 5;
+        func_8003426C(&this->dyna.actor, 0x4000, 0xFF, 0, 5);
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EN_PO_LAUGH2);
+        this->actionFunc = BgPoEvent_AmyPuzzle;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Event/func_808A69DC.s")
+void BgPoEvent_AmyPuzzle(BgPoEvent* this, GlobalContext* globalCtx) {
+    Vec3f pos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Event/func_808A6A94.s")
+    if (sPuzzleState == 0xF) {
+        pos.x = this->dyna.actor.posRot.pos.x - 5.0f;
+        pos.y = Math_Rand_CenteredFloat(120.0f) + this->dyna.actor.posRot.pos.y;
+        pos.z = Math_Rand_CenteredFloat(120.0f) + this->dyna.actor.posRot.pos.z;
+        EffectSsDeadDb_Spawn(globalCtx, &pos, &sZeroVec, &sZeroVec, 170, 0, 200, 255, 100, 170, 0, 255, 0, 1, 9, true);
+    } else if (sPuzzleState == 0x20) {
+        Actor_Kill(&this->dyna.actor);
+    } else {
+        DECR(this->timer);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Event/func_808A6BC8.s")
+s32 BgPoEvent_NextPicture(BgPoEvent* this) {
```

Comment:
just like BlockCheck isnt an action but has the Block prefix, should this be `PaintingNext`?